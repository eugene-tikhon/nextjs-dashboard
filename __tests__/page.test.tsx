import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '../app/page'
//import { toMatchImageSnapshot } from 'jest-image-snapshot'

//expect.extend({ toMatchImageSnapshot });

describe('Page', () => {
  it('renders a heading logo', () => {
    render(<Page />)
 
    const headingLogo = screen.getByText('Acme')
 
    expect(headingLogo).toBeInTheDocument()
  });

  it('renders homepage unchanged', () => {
    const { container } = render(<Page />)
    expect(container).toMatchSnapshot()
  });
});