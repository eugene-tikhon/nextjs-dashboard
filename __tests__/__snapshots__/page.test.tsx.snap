// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders homepage unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col p-6"
  >
    <div
      class="flex h-20 shrink-0 items-end rounded-lg bg-blue-500 p-4 md:h-52"
    >
      <div
        class="className flex flex-row items-center leading-none text-white"
      >
        <svg
          aria-hidden="true"
          class="h-12 w-12 rotate-[15deg]"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p
          class="text-[44px]"
        >
          Acme
        </p>
      </div>
    </div>
    <div
      class="mt-4 flex grow flex-col gap-4 md:flex-row"
    >
      <div
        class="flex flex-col justify-center gap-6 rounded-lg bg-gray-50 px-6 py-10 md:w-2/5 md:px-20"
      >
        <p
          class="text-xl text-gray-800 md:text-3xl md:leading-normal"
        >
          <strong>
            Welcome to Acme.
          </strong>
           This is the example for the
           
          <a
            class="text-blue-500"
            href="https://nextjs.org/learn/"
          >
            Next.js Learn Course
          </a>
          , brought to you by Vercel.
        </p>
        <a
          class="flex items-center gap-5 self-start rounded-lg bg-blue-500 px-6 py-3 text-sm font-medium text-white transition-colors hover:bg-blue-400 md:text-base"
          href="/login"
        >
          <span>
            Log in
          </span>
           
          <svg
            aria-hidden="true"
            class="w-5 md:w-6"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
      </div>
      <div
        class="flex items-center justify-center p-6 md:w-3/5 md:px-28 md:py-12"
      >
        <img
          alt="Screenshots of the dashboard project showing desktop version"
          class="hidden md:block"
          data-nimg="1"
          data-testid="image"
          decoding="async"
          height="760"
          loading="lazy"
          src="/_next/image?url=%2Fhero-desktop.png&w=2048&q=75"
          srcset="/_next/image?url=%2Fhero-desktop.png&w=1080&q=75 1x, /_next/image?url=%2Fhero-desktop.png&w=2048&q=75 2x"
          style="color: transparent;"
          width="1000"
        />
        <img
          alt="Screenshot of the dashboard project showing mobile version"
          class="block md:hidden"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
    </div>
  </main>
</div>
`;
